src/app/InteractionModelEngine.h

```diff
@@ -132,6 +133,21 @@ class InteractionModelEngine : public Messaging::UnsolicitedMessageHandler,
      */
     CASESessionManager * GetCASESessionManager() const { return mpCASESessionMgr; }
 
+    // reporting::EngineDelegate implementation
+    Messaging::ExchangeManager * GetExchangeManagerForReportingEngine() override { return mpExchangeMgr; }
+    ObjectPool<ReadHandler, CHIP_IM_MAX_NUM_READS + CHIP_IM_MAX_NUM_SUBSCRIPTIONS> & GetReadHandlerPool() override
+    {
+        return mReadHandlers;
```

Comment:
Why is this public?  We shouldn't have public access to the pool.

Reply:
I will need to re-figure out what things are needed and what are not. Reporting engine seems to use the pool directly (which may be questionable in itself...not sure) and providing some generic class did not seem easy because of callback-based access into the pool. 

Will spend some time to see how this should look like. Overall this seems to have started very coupled by providing a friend relationship and direct pool manipulation.